packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2822,26 +2822,56 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // The call to fetch the snapshot is very expensive and not always needed.
         // It should only be done by the summarizerNode, if required.
         const snapshotTreeFetcher = async () => {
+            // When fetching from storage we will always get the latest version and do not use the ackHandle.
             const fetchResult = await this.fetchSnapshotFromStorage(
-                ackHandle,
-                summaryLogger,
+             null,
+             summaryLogger,
+             {
+                 eventName: "RefreshLatestSummaryGetSnapshot",
+                 ackHandle,
+                 summaryRefSeq,
+                 fetchLatest: true,
+             });
+             return fetchResult.snapshotTree;
+         };
+         const result = await this.summarizerNode.refreshLatestSummary(
+             proposalHandle,
+             summaryRefSeq,
+             snapshotTreeFetcher,
+             readAndParseBlob,
+             summaryLogger,
+         );
+
+         // In case we had to retrieve the latest snapshot and it is different than summaryRefSeq,
+         // wait for the delta manager to catch up.
+         if (result.latestSummaryUpdated === true && result.wasSummaryTracked === false) {
+            assert(result.snapshot !== undefined, "should have tree result");
+            const latestSnapshotRefSeq = await seqFromTree(result.snapshot, readAndParseBlob);
+            if (latestSnapshotRefSeq !== summaryRefSeq) {
+                summaryLogger.sendTelemetryEvent(
```

Comment:
I think we should never get here, so it sending telemetry event does not feel appropriate. It should be fail fast in some sort (maybe assert, maybe more explicit container closure). 
I'd go for assert / exception, but follow up and make sure that HandleSummaryAckError event in handleSummaryAcks() actually closes this container. I believe I've opened a task about it, because we already had a case where assert in this flow was basically doing nothing (other than logging)

Reply:
Snapshot will always have a valid value - I do not see how it can not have. The assert was just to allow the call on 2849 to work as snapshot can be undefined and seqFromTree expects an actual snapshot. I could also use result.snapshot! or something. Is there a better way to do it?

Reply:
I'm not sure I follow. The way I read the code - it checks if seq number in summary op matches seq number in attributes in same summary. They should always match

Reply:
Oh I see - summaryRefSeq can be for different summary - the ack that we saw (that is in the past).
So maybe we should change it to be `>`, and also assert that latestSnapshotRefSeq >= summaryRefSeq ?